crates/binjs_io/src/entropy/model.rs

```diff
@@ -42,25 +42,21 @@ impl serde::ser::Serialize for FilesContaining {
 macro_rules! symbol {
     ( $me: ident, $table: ident, $description: expr, $path:expr, $value: expr ) => {
         {
-            let len = usize::min($me.dictionary.depth, $path.len()) + 1;
-            // Update predictions with a path length of 0..len.
-            for len in 0 .. len {
-                let tail = $path.tail(len);
-                $me.dictionary
-                    .$table
-                    .entry(tail, &$value)
-                    .value
-                    .and_modify(|instances| {
-                        *instances += 1 // We already have at least one association `tail` => `value`, increment.
-    //                }).or_insert(1);    // First time we associate `tail` => `value`, store 1.
-                    }).or_insert_with(|| {
-                        debug!(target: "entropy", "Injecting {} in context path: {:?}: {:?}",
-                            $description,
-                            tail,
-                            $value);
-                        1
-                    });    // First time we associate `tail` => `value`, store 1.
-            }
+            let tail = $path.tail($me.dictionary.depth);
```

Comment:
I had added this loop for an experiment, as a manner of estimating the cost/benefit for each possible path depth in the dictionary, without having to regenerate a dictionary each time.

For the moment, I'm not running the experiment, as I have no good way of interpreting the results, so I'm getting rid of that code. It might return later, once I have a better idea on the experiment.